dlib/image_saver/save_jxl_abstract.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2024  Davis E. King (davis@dlib.net), Adrià Arrufat
+// License: Boost Software License   See LICENSE.txt for the full license.
+#undef DLIB_SAVE_JXL_ABSTRACT_Hh_
+#ifdef DLIB_SAVE_JXL_ABSTRACT_Hh_
+
+#include "../image_processing/generic_image.h"
+#include "../pixel.h"
+#include <string>
+
+namespace dlib
+{
+
+// ----------------------------------------------------------------------------------------
+
+    template <
+        typename image_type
+        >
+    void save_jxl (
+        const image_type& img,
+        const std::string& filename,
+        float quality = 90
+    );
+    /*!
+        requires
+            - image_type == an image object that implements the interface defined in
+              dlib/image_processing/generic_image.h or a matrix expression
+            - image.size() != 0
+            - quality >= 0
+        ensures
+            - writes the image to the file indicated by filename in the JPEG XL format.
+            - image[0][0] will be in the upper left corner of the image.
+            - image[image.nr()-1][image.nc()-1] will be in the lower right corner of the
+              image.
+            - This routine can save images containing any type of pixel.  However,
+              save_jxl() can only natively store rgb_pixel, and rgb_alpha_pixel pixel types.
```

Comment:
Doesn't it also save grayscale too?

Reply:
Right! You can see I reused the WebP saver :sweat_smile:

Reply:
Yeah, that one might as well be private in the .cpp file too.  But :shrug:  :D

Reply:
I will add those methods in another PR, maybe I can add `nr` and `nc` too to jpeg and png loaders, too.